src/uct/ib/base/ib_iface.c

```diff
@@ -821,7 +889,8 @@ UCS_CLASS_INIT_FUNC(uct_ib_iface_t, uct_ib_iface_ops_t *ops, uct_md_h md,
 
     status = uct_ib_device_select_gid_index(dev, self->config.port_num,
                                             ib_md->config.gid_index,
-                                            &self->config.gid_index);
+                                            &self->config.gid_index,
+                                            &self->roce_ver);
```

Comment:
it seems redundant that in line 898 we also query for gid information and roce version
can we remove the returning of self->roce_ver from uct_ib_device_select_gid_index, and move it to uct_ib_device_query_gid() instead?

Reply:
uct_ib_device_select_gid_index already calls what used to be uct_ib_device_set_roce_gid_index where it goes over the roce versions to set the gid_index. so since it already has the roce versions there, wont it be better to use it?

Reply:
remove is_roce_v2 from uct_ib_device_query_gid. instead set is_roce_v2 by checking self->roce_ver if port is roce